crates/services/compression/src/service.rs

```diff
@@ -29,16 +32,24 @@ use fuel_core_services::{
     ServiceRunner,
     StateWatcher,
 };
-use futures::FutureExt;
+use fuel_core_storage::transactional::WriteTransaction;
+use futures::{
+    FutureExt,
+    StreamExt,
+};
 
 /// The compression service.
 /// Responsible for subscribing to the l2 block stream,
 /// perform compression of the block, and store the compressed block.
 /// We don't need to share any data between this task and anything else yet(?)
 /// perhaps we want to expose a way to get the registry root
-pub struct CompressionService<S> {
+pub struct CompressionService<B, S, CH> {
+    /// The block source.
+    block_source: B,
     /// The block stream.
     block_stream: crate::ports::block_source::BlockStream,
+    /// The canonical height getter.
+    canonical_height: CH,
```

Comment:
It looks like you only use it during `into_task` function. So, it would be nice if we had either uninitialized and initialized types, or if you passed `canonical_height` via the `TaskParams`(maybe you can just pass the height)

Reply:
The same about `block_source`

Reply:
addressed in e9d5b0903b.

I prefer to keep the `canonical_height` as a getter and not directly the height to ensure that we get the height before we start the sync loop. i don't want to assume here that no blocks will be produced before the service starts. (even though that is the case _now_ but might not always be in the future if we change the behaviour)